lib/semigrp.gi

```diff
@@ -540,10 +540,14 @@ InstallMethod( AsSemigroup,
 
 InstallMethod( AsSemigroup,
     "generic method for collections",
-    [ IsCollection ],
+    [ IsMultiplicativeElementCollection ],
```

Comment:
Thank you for this explanation. You are correct that this PR breaks your example.

The motivation for this change was that I generalised the declaration of `IsAssociative` from `IsMagma` to `IsMultiplicativeElementCollection`. Since this `AsSemigroup` method now uses `IsAssociative` in turn, I thought that these should agree, as otherwise the `AsSemigroup` method would give a 'no method found' error when it got to the call to `IsAssociative`.

Should I have instead declared `IsAssociative` for `IsCollection`? That means that `IsAssociative( C )` for a collection of (say) digraphs would give a 'no method found for *`' error, but I suppose that is reasonable? This seems like the right approach to me.